pkg/app/AppService.go

```diff
@@ -2195,47 +2281,142 @@ func (impl *AppServiceImpl) getConfigMapAndSecretJsonV2(appId int, envId int, pi
 	var configMapJsonEnv string
 	var secretDataJsonEnv string
 	var err error
+	var scope resourceQualifiers.Scope
+	var entitiesForCM []repository6.Entity
+	var entitiesForCS []repository6.Entity
+	var entityToVariablesCM map[repository6.Entity][]string
+	var entityToVariablesCS map[repository6.Entity][]string
+	var resolvedTemplateCM string
+	var variableMapCM map[string]string
+	var resolvedTemplateCS string
+	var variableMapCS map[string]string
 	//var configMapJsonPipeline string
 	//var secretDataJsonPipeline string
 
 	merged := []byte("{}")
-	if deploymentWithConfig == bean.DEPLOYMENT_CONFIG_TYPE_LAST_SAVED {
-		configMapA, err := impl.configMapRepository.GetByAppIdAppLevel(appId)
+	if cMCSJson.DeploymentWithConfig == bean.DEPLOYMENT_CONFIG_TYPE_LAST_SAVED {
+		//cMCSJson.envOverride = &chartConfig.EnvConfigOverride{}
+
+		//VARIABLE different cases for variable resolution
+		env, err := impl.envRepository.FindById(cMCSJson.EnvId)
+		if err != nil {
+			impl.logger.Errorw("unable to find env", "err", err)
+			return nil, err
+		}
```

Comment:
why is this call needed, I believe it have been already made in the trace and all the required values are passed

Reply:
done